src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -340,5 +412,349 @@ private static ICollection<object> PopulateHashTableFromJArray(JArray list, out
+                    {
+                        var errorRecord = new ErrorRecord(exception, "NonStringKeyInDictionary", ErrorCategory.InvalidOperation, dict);
+                        context.Cmdlet.ThrowTerminatingError(errorRecord);
+                    }
+                    else
+                    {
+                        throw exception;
+                    }
+                }
+
+                result.Add(name, ProcessValue(entry.Value, depth + 1, in context));
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Return an alternate representation of the specified collection that serializes the same JSON, except
+        /// that any contained properties that cannot be evaluated are treated as having the value null.
+        /// </summary>
+        private static object ProcessEnumerable(IEnumerable enumerable, int depth, in ConvertToJsonContext context)
+        {
+            List<object> result = new List<object>();
+
+            foreach (object o in enumerable)
+            {
+                result.Add(ProcessValue(o, depth + 1, in context));
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Return an alternate representation of the specified aggregate object that serializes the same JSON, except
+        /// that any contained properties that cannot be evaluated are treated as having the value null.
+        ///
+        /// The result is a dictionary in which all public fields and public gettable properties of the original object
+        /// are represented.  If any exception occurs while retrieving the value of a field or property, that entity
+        /// is included in the output dictionary with a value of null.
+        /// </summary>
+        private static object ProcessCustomObject<T>(object o, int depth, in ConvertToJsonContext context)
+        {
+            Dictionary<string, object> result = new Dictionary<string, object>();
+            Type t = o.GetType();
+
+            foreach (FieldInfo info in t.GetFields(BindingFlags.Public | BindingFlags.Instance))
+            {
+                if (!info.IsDefined(typeof(T), true))
+                {
+                    object value;
+                    try
+                    {
+                        value = info.GetValue(o);
+                    }
+                    catch (Exception)
+                    {
+                        value = null;
+                    }
+
+                    result.Add(info.Name, ProcessValue(value, depth + 1, in context));
+                }
+            }
+
+            foreach (PropertyInfo info2 in t.GetProperties(BindingFlags.Public | BindingFlags.Instance))
+            {
+                if (!info2.IsDefined(typeof(T), true))
+                {
+                    MethodInfo getMethod = info2.GetGetMethod();
+                    if ((getMethod != null) && (getMethod.GetParameters().Length <= 0))
```

Comment:
Can getMethod.GetParameters() return null?

Reply:
I don't think it can return null.